05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+It became clear to us that we must chose a mathematical representation of rules, since agents would be able to much mroe easily inspect mathematics over a text or functional representation of rules.
+The mathematics we settled on was Matrices, and the world of linear algebra that surrounds them.
+\linebreak
+The premise of this choice was that if a rule could be represented by matrices, then agents could look up every element of the matrix without difficulty. We'd effectively be able to expose the entire operation to the agent without reducing the scope of our rules too much.
+To give an example of these matrix based rules, let us consider a simple rule first. 
+"If you are expected to pay x amount of tax, the amount of tax you pay must be x"
+\par
+Quite obviously this is a very simple rule, however it allows us to form a basis for how to represent Rules as matrices.
+If our goal is simply to ensure that an agent pays the expected amount of tax, a basic mathematical operation we can perform is subtract the actual tax paid from the expected.
+If we get 0, we know this agent has adhered to this rule. 
+\[
+  \textrm{let expected tax} = x   
+\]
+\[
+    \textrm{let actual paid tax} = y    
+\]
+\[
+    \textrm{We want    }
+      y - x = 0
+\]
+To turn this into a matrix calculation, we simply have to introduce some trivial coefficients.
+\[
+      -1*x +
+       1*y = 0
+\]
+Now we can write this as a matrix calculation.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    = 0  
+\]
+
+We have now encoded this simply rule as a matrix product. An agent can input their versions of x and y, and perform the calculation.
+The agent can then check if their result is 0, if so they are compliant and can pay $y$ tax. Let us revisit our rule, and evolve it a little.
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x"
+\linebreak
+\linebreak
+We can rewrite the matrix equation as an inequality.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    >= 0  
+\]
+The addition of the "at least" adds a little bit of complexity to our matrix schema. In that we may not be able to simply check for equality to a constant, and may instead need to check for a certain condition.
+To make these comparisons easier, we have decided to ensure that whatever condition, equality or inequality, we would compare to 0.
+However if our tax rule looked like this (for whatever reason):
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x + 5"
+\linebreak
+\linebreak
+Then it would be impossible to capture that using the current matrix calculation, so to capture these cases we introduce a constant to the input list, as below.
```

Comment:
```suggestion
In this case, it would be impossible to capture this rule using the previously suggested matrix scheme. Therefore, to capture such cases of rules, we introduce a \emph{constant} to the input list, as below.
```